### ISSUE TITLE
Warnings at compile time

### ISSUE THREAD

**Issue by shg** [AndreaLombardo/L298N#7]
I get the following warnings at compile time. The type of the argument of L298NX2::run seems wrong?

```text
/Users/shg/Documents/Arduino/libraries/L298N/src/L298NX2.cpp: In member function 'void L298NX2::run(uint8_t)':
/Users/shg/Documents/Arduino/libraries/L298N/src/L298NX2.cpp:260:24: warning: invalid conversion from 'uint8_t {aka unsigned char}' to 'L298N::Direction' [-fpermissive]
   _motorA.run(direction);
                        ^
In file included from /Users/shg/Documents/Arduino/libraries/L298N/src/L298NX2.h:5:0,
                 from /Users/shg/Documents/Arduino/libraries/L298N/src/L298NX2.cpp:7:
/Users/shg/Documents/Arduino/libraries/L298N/src/L298N.h:28:9: note:   initializing argument 1 of 'void L298N::run(L298N::Direction)'
    void run(L298N::Direction direction);
         ^~~
/Users/shg/Documents/Arduino/libraries/L298N/src/L298NX2.cpp:261:24: warning: invalid conversion from 'uint8_t {aka unsigned char}' to 'L298N::Direction' [-fpermissive]
   _motorB.run(direction);
                        ^
In file included from /Users/shg/Documents/Arduino/libraries/L298N/src/L298NX2.h:5:0,
                 from /Users/shg/Documents/Arduino/libraries/L298N/src/L298NX2.cpp:7:
/Users/shg/Documents/Arduino/libraries/L298N/src/L298N.h:28:9: note:   initializing argument 1 of 'void L298N::run(L298N::Direction)'
    void run(L298N::Direction direction);
         ^
```

```diff
--- L298NX2.cpp.orig	2020-09-02 21:47:44.000000000 +0900
+++ L298NX2.cpp	2020-09-02 21:42:15.000000000 +0900
@@ -255,7 +255,7 @@
   _motorB.backward();
 }

-void L298NX2::run(uint8_t direction)
+void L298NX2::run(L298N::Direction direction)
 {
   _motorA.run(direction);
   _motorB.run(direction);
--- L298NX2.h.orig	2020-09-02 21:48:02.000000000 +0900
+++ L298NX2.h	2020-09-02 21:42:33.000000000 +0900
@@ -63,7 +63,7 @@
    void backward();
    void backwardFor(unsigned long delay, CallBackFunction callback);
    void backwardFor(unsigned long delay);
-   void run(uint8_t direction);
+   void run(L298N::Direction direction);
    void runFor(unsigned long delay, L298N::Direction direction);
    void runFor(unsigned long delay, L298N::Direction direction, CallBackFunction callback);
    void stop();
```

**Comment by AndreaLombardo** [comment]
Hi, sorry for the long delay.
Can you share your sketch? I re tested the example sketch and none of them seem to generate this issue.
Thanks

**Comment by jsodeman** [comment]
I'm getting the same thing from the L298N-Simple example running on a Wemos D1 mini 8266 using Platform.io.

**Comment by shg** [comment]
@AndreaLombardo I can't test it now, but as far as I remember, I also got the warning when running example sketches.

Based on the code, the type of arguments to L298NX2::run should be L298N::Direction instead of uint8_t because the methods called from within L298NX2::run, _motorA.run and _motorB.run, both expect L298N::Direction type as the argument. I confirmed the warnings disappear with the fix I included in the above comment.
